### PR TITLE
Checkstyle: remove use of Parquet Preconditions and Apache Commons IOUtils

### PR DESCRIPTION
**Pull request by jackye1995**
Tested checkstyle works:

```
> Task :iceberg-spark:iceberg-spark-3.2:checkstyleMain
[ant:checkstyle] [ERROR] /projects/iceberg/spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/data/vectorized/ColumnarBatchReader.java:26:1: Use Preconditions from bundled-guava module instead. [GuavaClassesInParquet]

> Task :iceberg-aliyun:checkstyleTest FAILED
[ant:checkstyle] [ERROR] /projects/iceberg/aliyun/src/test/java/org/apache/iceberg/aliyun/oss/mock/AliyunOSSMockLocalController.java:172: Prefer Guava's ByteStreams.copy to avoid charset/stream closing issues. [RegexpSinglelineJava]
```

### CODE REVIEW

File: .baseline/checkstyle/checkstyle.xml [hunk 1]
```diff
@@ -361,6 +366,10 @@
             <property name="format" value="\bIOUtils\.toString\("/>
             <property name="message" value="Prefer Guava''s [CharStreams,Files,Resources].toString to avoid charset/stream closing issues."/>
         </module>
+        <module name="RegexpSinglelineJava">
+            <property name="format" value="\bIOUtils\.copy\("/>
+            <property name="message" value="Prefer Guava''s ByteStreams.copy to avoid charset/stream closing issues."/>
```
Comment (kbendick) [inline .baseline/checkstyle/checkstyle.xml:371]: 👍 . Thanks for adding this rule and the nice explanation.

**Review by kbendick — APPROVED** [review]
Looks good to me. Thanks @jackye1995!

File: .baseline/checkstyle/checkstyle.xml (hunk 1 above)
Comment (nastra) [reply .baseline/checkstyle/checkstyle.xml:371]: nit: contains 2x '' instead of a single '

**Review by nastra — APPROVED** [review]
LGTM

File: .baseline/checkstyle/checkstyle.xml (hunk 1 above)
Comment (jackye1995) [reply .baseline/checkstyle/checkstyle.xml:371]: `''` is used in the checkstyle file to escape `'`

**Comment by jackye1995** [comment]
Thanks for the review! @kbendick @nastra

**Pull request merged**